examples/jf-admin-app/linux/JFADatastoreSync.cpp

```diff
@@ -0,0 +1,232 @@
+#include <lib/support/logging/CHIPLogging.h>
+
+#include <controller/CHIPDeviceController.h>
+#include <controller/CommissionerDiscoveryController.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::Controller;
+using namespace chip::app::Clusters::JointFabricDatastore;
+
+extern DeviceCommissioner * GetDeviceCommissioner();
+
+constexpr uint8_t kJFAvailableShift = 0;
+constexpr uint8_t kJFAdminShift     = 1;
+constexpr uint8_t kJFAnchorShift    = 2;
+constexpr uint8_t kJFDatastoreShift = 3;
+
+static constexpr EndpointId kJFDatastoreClusterEndpointId = 1;
+
+JFADatastoreSync JFADatastoreSync::sJFDS;
+
+template <typename T>
+class DevicePairedCommand
+{
+public:
+    struct CallbackContext
+    {
+        chip::NodeId nodeId;
+        T objectToWrite;
+        std::function<void()> onComplete;
+
+        CallbackContext(chip::NodeId nId, T object, std::function<void()> onComp) :
+            nodeId(nId), objectToWrite(object), onComplete(onComp)
+        {}
+    };
+    DevicePairedCommand(
+        chip::NodeId nodeId, T object, std::function<void()> onComplete = []() {}) :
+        mOnDeviceConnectedCallback(OnDeviceConnectedFn, this), mOnDeviceConnectionFailureCallback(OnDeviceConnectionFailureFn, this)
+    {
+        mContext = std::make_shared<CallbackContext>(nodeId, object, onComplete);
+    }
+
+    static void OnDeviceConnectedFn(void * context, chip::Messaging::ExchangeManager & exchangeMgr,
+                                    const chip::SessionHandle & sessionHandle)
+    {
+        auto * pairingCommand = static_cast<DevicePairedCommand *>(context);
+        auto cbContext        = pairingCommand->mContext;
+
+        if (pairingCommand)
+        {
+            ChipLogProgress(DeviceLayer, "OnDeviceConnectedFn - Syncing device with node id: " ChipLogFormatX64,
+                            ChipLogValueX64(cbContext->nodeId));
+
+            chip::Controller::ClusterBase cluster(exchangeMgr, sessionHandle, kJFDatastoreClusterEndpointId);
+
+            // Static LUT for mapping type T to attribute ID
+            AttributeId attributeId = 0;
+
+            if constexpr (std::is_same_v<T,
+                                         app::Clusters::JointFabricDatastore::Structs::DatastoreEndpointGroupIDEntryStruct::Type>)
+            {
+                attributeId = Attributes::EndpointGroupIDList::Id;
+            }
+            else if constexpr (std::is_same_v<T,
+                                              app::Clusters::JointFabricDatastore::Structs::DatastoreNodeKeySetEntryStruct::Type>)
+            {
+                attributeId = Attributes::NodeKeySetList::Id;
+            }
+            else if constexpr (std::is_same_v<
+                                   T, app::Clusters::JointFabricDatastore::Structs::DatastoreEndpointBindingEntryStruct::Type>)
+            {
+                attributeId = Attributes::EndpointBindingList::Id;
+            }
```

Comment:
In the case of DatastoreEndpointBindingEntryStruct, the operation to send is:

IM Type: AttributeWrite
EndpointId=0 (root endpoint)
ClusterId=0x001E (bindings cluster)
AttributeId=0x00 (binding)
Arg0=list of all bindings for this node from datastore